indico/modules/events/editing/controllers/backend/management.py

```diff
@@ -145,3 +146,25 @@ def _process_POST(self, editable_types):
         editable_types_names = [t.name for t in editable_types]
         editing_settings.set(self.event, 'editable_types', editable_types_names)
         return '', 204
+
+
+class RHEditableTypePrincipals(RHEditingManagementBase):
+    def _process_args(self):
+        RHEditingManagementBase._process_args(self)
+        self.editable_type = EditableType[request.view_args['type']]
+
+    def _process_GET(self):
+        principals = {p.principal for p in self.event.acl_entries
+                      if p.has_management_permission(self.editable_type.name + '_editing', explicit=True)}
+        return EditableTypePrincipalsSchema().jsonify({'principals': principals})
```

Comment:
I'd just return a list here:

    return jsonify([p.principal.identifier for p in principals])

Reply:
Wouldn't this break in case of `CategoryRole` for example?

Reply:
No, any principal class has an `identifier` property that returns a string like `User:123` or `CategoryRole:123`.

This is actually what PrincipalList does when serializing:

```python
    def _serialize(self, value, attr, obj):
        return [x.identifier for x in value]
```

Reply:
Ah, I see. I was already accessing `p.principal`